src/main/java/de/tum/cit/aet/artemis/core/web/FileResource.java

```diff
@@ -207,7 +217,16 @@ public ResponseEntity<byte[]> getMarkdownFileForConversation(@PathVariable Long
         // TODO: Improve the access check
         log.debug("REST request to get file for markdown in conversation: File {} for conversation {} in course {}", filename, conversationId, courseId);
         sanitizeFilenameElseThrow(filename);
-        return buildFileResponse(FilePathService.getMarkdownFilePathForConversation(courseId, conversationId), filename, true);
+
+        var path = FilePathService.getMarkdownFilePathForConversation(courseId, conversationId);
+
+        var fileUpload = fileUploadService.findByPath("/api/files/courses/" + courseId + "/conversations/" + conversationId + "/" + filename);
```

Comment:
Do I understand it correctly, that you first save the file, and then retrieve it again from the database? Why is this necessary?

Reply:
Hey, thanks for your review! 😄 Here we just take the sanitized filename that the user queries and we query the db if the file upload entity exists. The saving happens elsewhere. We do this in order to retrieve the original filename. The idea here is to not store files with original filenames, but still make them downloadable with them. We do this to kind of navigate around some security headaches that filenames can create when storing them.